Objective-C/CBLQueryDataSource.m

```diff
@@ -32,7 +32,8 @@ - (instancetype) initWithDataSource: (id)source as: (nullable NSString*)alias {
 }
 
 - (id) asJSON {
-    return _alias ?  @{@"AS": _alias} : @{ };
+    NSString* columnName = [self columnName];
+    return columnName ? @{ @"AS" : columnName } : @{ };
```

Comment:
Instead of allowing @ { } which is invalid now, we can assert(columnName) instead.